src/main/java/teammates/storage/sqlapi/FeedbackSessionLogsDb.java

```diff
@@ -0,0 +1,88 @@
+package teammates.storage.sqlapi;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+
+import teammates.common.util.HibernateUtil;
+import teammates.storage.sqlentity.FeedbackSession;
+import teammates.storage.sqlentity.FeedbackSessionLog;
+import teammates.storage.sqlentity.Student;
+
+import jakarta.persistence.criteria.CriteriaBuilder;
+import jakarta.persistence.criteria.CriteriaQuery;
+import jakarta.persistence.criteria.Join;
+import jakarta.persistence.criteria.Predicate;
+import jakarta.persistence.criteria.Root;
+
+/**
+ * Handles CRUD operations for feedback session logs.
+ *
+ * @see FeedbackSessionLog
+ */
+public final class FeedbackSessionLogsDb extends EntitiesDb {
+
+    private static final FeedbackSessionLogsDb instance = new FeedbackSessionLogsDb();
+
+    private FeedbackSessionLogsDb() {
+        // prevent initialization
+    }
+
+    public static FeedbackSessionLogsDb inst() {
+        return instance;
+    }
+
+    /**
+     * Gets the feedback session logs as filtered by the given parameters ordered by
+     * ascending timestamp. Logs with the same timestamp will be ordered by the
+     * student's email.
+     *
+     * @param courseId            Can be null
+     * @param studentEmail        Can be null
+     * @param feedbackSessionName Can be null
+     */
+    public List<FeedbackSessionLog> getOrderedFeedbackSessionLogs(String courseId, String studentEmail,
+            String feedbackSessionName, Instant startTime, Instant endTime) {
+
+        assert startTime != null;
+        assert endTime != null;
+
+        CriteriaBuilder cb = HibernateUtil.getCriteriaBuilder();
+        CriteriaQuery<FeedbackSessionLog> cr = cb.createQuery(FeedbackSessionLog.class);
+        Root<FeedbackSessionLog> root = cr.from(FeedbackSessionLog.class);
+        Join<FeedbackSessionLog, FeedbackSession> feedbackSessionJoin = root.join("feedbackSession");
+        Join<FeedbackSessionLog, Student> studentJoin = root.join("student");
+
+        List<Predicate> predicates = new ArrayList<>();
+
+        if (courseId != null) {
+            predicates.add(cb.equal(feedbackSessionJoin.get("course").get("id"), courseId));
+        }
```

Comment:
Based on the design, I think course is mandatory. The instructor can only query the logs for 1 course. If somehow no course is provided, we shouldn't proceed with the query.

Reply:
Good catch! The constraint is currently enforced by the get action. But disallowing this will help to simplify the test cases a lot.